src/order-with-promo/tests.t

```diff
@@ -0,0 +1,123 @@
+Sandwich tests
+  $ node ./output/src/order-with-promo/SandwichTests.mjs | sed '/duration_ms/d'
```

Comment:
What's the reasoning behind running node inside cram tests and snapshotting the output?

Looks very fragile (every test changes you need to update the snapshot)

Reply:
I just didn't think of a better way to do it. Can you suggest an alternative?

Reply:
Run node directly, not via dune is an option. But also, not use cram test and run node inside dune via an action

Reply:
I think you're suggesting how it's set up in melange-fest: https://github.com/ahrefs/melange-fest/blob/2b805fddfdcfcb659215a02bc837549977ebbe07/test/dune#L14-L20

The original version did use rules, but I didn't like it for two reasons:

- It took a lot more space to explain, at a point in the book when the reader isn't that familiar with Dune yet (but we need to introduce some kind of testing to make it easy for users to check if they did the exercises correctly)
- The actual experience of using it was a bit worse than cram tests. Maybe it's just personal taste, but I like using cram tests for this project more than dune rules. One nice thing, for example, is that it doesn't print anything at all if no tests failed and no tests were added/removed.

Like you said, though, it is more fragile, but maybe in this project that's an asset instead of a liability? 🤔

Reply:
No no, I mean directly run node in the console

Reply:
I show the reader how to do that in the [Sandwich Tests](https://react-book.melange.re/sandwich-tests/#add-test-npm-script). They're free to stick with that approach if they like, but I personally don't like it better than the cram test approach I adopt in later chapters.